src/components/widgets/PAutoComplete.js

```diff
@@ -1,26 +1,71 @@
-import { PInput, PListItem, PList } from '../elements'
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { PInput, PListItem, PList, PLoading, PNoData } from '../elements'
+
 export default (props) => {
+  const {
+    value = '',
+    width,
+    size,
+    suggestions = [],
+    loading,
+    onChange = () => null,
+    onSelect = () => null
+  } = props
+
+  const [interalValue, setInternalValue] = useState(value)
+  const [interalSuggestions, setInternalSuggestions] = useState(suggestions)
+  const [showSuggestions, setShowsuggestions] = useState(false)
+  const [isNoData, setIsNoData] = useState(false)
```

Comment:
Avoid to use redundant state. isNoData can be computed by checking suggestion list length